pgmpy/base/DAG.py

```diff
@@ -1583,6 +1583,139 @@ def edge_strength(self, data, edges=None):
 
         return strengths
 
+    def validate(self, data, metrics=None, **kwargs):
+
+        from sklearn.metrics import f1_score
+
+        from pgmpy.estimators.CITests import chi_square
+        from pgmpy.metrics.metrics import (
+            correlation_score,
+            fisher_c,
+            implied_cis,
+            permutation_test,
+            structure_score,
+        )
+
+        # all validation metrics
+        all_metrics = [
+            "correlation",
+            "aic",
+            "bic",
+            "fisher_c",
+            "implied_cis",
+            "permutation_test",
+        ]
+
+        # a normal validate call would provide all metric results
+        if metrics is None:
+            metrics = all_metrics
+
+        # Dictionary for holding default parameters
+        params = {
+            "correlation_test": "chi_square",
```

Comment:
This needs to be automatically detected based on the data. Have a look at: https://github.com/pgmpy/pgmpy/blob/dev/pgmpy/estimators/CITests.py#L14